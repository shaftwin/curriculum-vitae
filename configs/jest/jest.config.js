const path = require('path');

module.exports = {
  rootDir: '../../',
  testEnvironment: 'jsdom',
  testMatch: ['**/*.test.js'],
  setupFilesAfterEnv: ['./node_modules/@testing-library/jest-dom'],
  moduleDirectories: [path.resolve(__dirname, '../../'), 'node_modules'],
  setupFiles: [
    './configs/jest/mock.js',
    './configs/cucumber/jest-cucumber-config.js',
  ],
  snapshotSerializers: ['@emotion/jest/serializer'],
};
