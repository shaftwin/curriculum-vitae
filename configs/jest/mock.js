jest.mock('react-i18next', () => ({
  withTranslation: () => (Component) => {
    // eslint-disable-next-line no-param-reassign
    Component.defaultProps = { ...Component.defaultProps, t: () => '' };
    return Component;
  },
  // this mock makes sure any components using the
  // translate hook can use it without a warning being shown
  useTranslation: () => {
    return {
      t: (str) => str,
      i18n: {
        changeLanguage: () => new Promise(() => {}),
      },
    };
  },
}));
