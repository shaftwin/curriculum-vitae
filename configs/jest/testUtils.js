// import { MockedProvider } from '@apollo/client/testing';
import { ThemeProvider } from '@emotion/react';
import { render } from '@testing-library/react';
import React from 'react';
import theme from '../../utils/theme';

const AllTheProviders = ({ children }) => {
  return <ThemeProvider theme={theme}>{children}</ThemeProvider>;
};

const customRender = (ui, options) =>
  render(ui, { wrapper: AllTheProviders, ...options });

// re-export everything
export * from '@testing-library/react';

// override render method
export { customRender as render };
