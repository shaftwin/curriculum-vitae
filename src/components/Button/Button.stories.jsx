/* eslint-disable react/jsx-props-no-spreading */
import React from 'react';
// eslint-disable-next-line import/no-extraneous-dependencies
import { actions } from '@storybook/addon-actions';
import Filter from '../../../assets/svg/filter';
import Button from './index';

/* More on default export:
https://storybook.js.org/docs/react/writing-stories/introduction#default-export
 */
export default {
  title: 'Components/Button',
  component: Button,
};

/* More on component templates:
https://storybook.js.org/docs/react/writing-stories/introduction#using-args
*/

const Template = (args) => <Button onclick={actions('onClick')} {...args} />;

// More on args: https://storybook.js.org/docs/react/writing-stories/args

export const FillNormal = Template.bind({});
FillNormal.args = {
  label: 'Button',
};

export const FillWithIcon = Template.bind({});
FillWithIcon.args = {
  icon: <Filter />,
  label: 'Button',
};

export const StrokeNormal = Template.bind({});
StrokeNormal.args = {
  label: 'Button',
  buttonType: 'stroke',
};

export const StrokeWithIcon = Template.bind({});
StrokeWithIcon.args = {
  icon: <Filter />,
  label: 'Button',
  buttonType: 'stroke',
};

export const SmallWithIcon = Template.bind({});
SmallWithIcon.args = {
  icon: <Filter />,
  label: 'Button',
  size: 'small',
};
