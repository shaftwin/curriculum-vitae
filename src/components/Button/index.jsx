import React from 'react';
import PropTypes from 'prop-types';
import { StyledButton, IconContainer } from './styled';

const Button = ({
  label,
  size,
  onClick,
  disabled,
  icon,
  iconPosition,
  className,
  templateColor,
  buttonType,
  ...props
}) => {
  return (
    <StyledButton
      className={className}
      disabled={disabled}
      size={size}
      onClick={onClick}
      icon={icon}
      iconPosition={iconPosition}
      templateColor={templateColor}
      buttonType={buttonType}
      // eslint-disable-next-line react/jsx-props-no-spreading
      {...props}
    >
      {!!icon && (
        <IconContainer data-testid="iconContainer">{icon}</IconContainer>
      )}
      {label}
    </StyledButton>
  );
};

Button.propTypes = {
  /**
   * Button label
   */
  label: PropTypes.string.isRequired,
  /**
   * Optional button size
   */
  size: PropTypes.oneOf(['extraSmall', 'small', 'medium']),
  /**
   * Optional click handler
   */
  onClick: PropTypes.func,
  /**
   * Optional disabled
   */
  disabled: PropTypes.bool,
  /**
   * Optional icon
   */
  icon: PropTypes.element,
  /**
   * Optional iconPosition
   */
  iconPosition: PropTypes.oneOf(['left', 'right']),
  /**
   * Optional custom style
   */
  className: PropTypes.string,
  /**
   * Optional template color
   */
  templateColor: PropTypes.oneOf(['primary', 'secondary', 'tertiary', 'white']),
  /**
   * Optional buttonType
   */
  buttonType: PropTypes.oneOf(['fill', 'stroke', 'text']),
};

Button.defaultProps = {
  onClick: () => null,
  disabled: false,
  icon: null,
  iconPosition: 'left',
  size: 'medium',
  templateColor: 'primary',
  buttonType: 'fill',
};

export default Button;
