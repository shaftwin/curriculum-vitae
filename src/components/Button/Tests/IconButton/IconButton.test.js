import * as React from 'react';
import { screen } from '@testing-library/react';

import { defineFeature, loadFeature } from 'jest-cucumber';
import { render } from '../../../../../configs/jest/testUtils';
import Filter from '../../../../../assets/svg/filter';
import Button from '../..';

const feature = loadFeature(
  'src/components/Button/Tests/IconButton/IconButton.feature',
);

const flexDirection = (element) =>
  window.getComputedStyle(element).flexDirection;

defineFeature(feature, (test) => {
  test('IconButton has icon and the icon is at the left by default', ({
    given,
    then,
    and,
  }) => {
    let container;
    let icon;
    let button;
    given('Using the default iconLocation', () => {
      container = render(
        <Button label="testLabel" disabled={true} icon={<Filter />} />,
      ).container;
      icon = screen.queryByTestId(/iconContainer/i);
      button = screen.getByText('testLabel');
      expect(container).toMatchSnapshot();
    });

    then('The icon should exist', () => {
      expect(icon).not.toBeNull();
    });

    and('The icon should be at the left', () => {
      expect(flexDirection(button)).toBe('row');
    });
  });

  test('IconButton has icon at the right', ({ given, then, and }) => {
    let container;
    let icon;
    let button;
    given('iconLocation set to right', () => {
      container = render(
        <Button
          label="testLabel"
          iconPosition="right"
          disabled={true}
          icon={<Filter />}
        />,
      ).container;
      icon = screen.queryByTestId(/iconContainer/i);
      button = screen.getByText('testLabel');
      expect(container).toMatchSnapshot();
    });

    then('The icon should exist', () => {
      expect(icon).not.toBeNull();
    });

    and('The icon is at should be at the right', () => {
      expect(flexDirection(button)).toBe('row-reverse');
    });
  });
});
