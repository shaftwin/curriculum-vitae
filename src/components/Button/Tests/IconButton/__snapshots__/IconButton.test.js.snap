// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IconButton IconButton has icon and the icon is at the left by default 1`] = `
.emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: space-around;
  -ms-flex-pack: space-around;
  -webkit-justify-content: space-around;
  justify-content: space-around;
  min-width: 7.5rem;
  height: 2rem;
  border-radius: 0.438rem;
  cursor: pointer;
  font-weight: 700;
  font-size: 0.813rem;
  background-color: #00DBC7;
  border: 0;
  color: #120D2E;
}

.emotion-0:hover {
  background-color: #00C6B3;
  border: 0;
  color: #120D2E;
}

.emotion-0:disabled {
  cursor: default;
  background-color: #EBEEF0;
  border: 0;
  color: #888596;
}

.emotion-1 {
  margin-top: 4px;
}

<div>
  <button
    class="emotion-0"
    disabled=""
  >
    <div
      class="emotion-1"
      data-testid="iconContainer"
    >
      <svg
        fill="none"
        height="20"
        viewBox="0 0 20 20"
        width="20"
        xmlns="http://www.w3.org/2000/svg"
      >
        <mask
          fill="white"
          id="path-1-inside-1_1952_2686"
        >
          <path
            clip-rule="evenodd"
            d="M7.22703 3.72733C7.25685 3.58044 7.27251 3.42841 7.27251 3.27273C7.27251 3.11708 7.25686 2.96509 7.22705 2.81824H19.5455C19.7965 2.81824 20 3.02174 20 3.27278C20 3.52382 19.7965 3.72733 19.5455 3.72733H7.22703ZM2.77252 3.72733H0.454546C0.203508 3.72733 0 3.52382 0 3.27278C0 3.02174 0.203507 2.81824 0.454545 2.81824H2.7725C2.7427 2.96509 2.72705 3.11708 2.72705 3.27273C2.72705 3.42841 2.7427 3.58044 2.77252 3.72733Z"
            fill-rule="evenodd"
          />
        </mask>
        <path
          clip-rule="evenodd"
          d="M7.22703 3.72733C7.25685 3.58044 7.27251 3.42841 7.27251 3.27273C7.27251 3.11708 7.25686 2.96509 7.22705 2.81824H19.5455C19.7965 2.81824 20 3.02174 20 3.27278C20 3.52382 19.7965 3.72733 19.5455 3.72733H7.22703ZM2.77252 3.72733H0.454546C0.203508 3.72733 0 3.52382 0 3.27278C0 3.02174 0.203507 2.81824 0.454545 2.81824H2.7725C2.7427 2.96509 2.72705 3.11708 2.72705 3.27273C2.72705 3.42841 2.7427 3.58044 2.77252 3.72733Z"
          fill="#120D2E"
          fill-rule="evenodd"
        />
        <path
          d="M7.22703 3.72733L6.24702 3.52837L6.00362 4.72733H7.22703V3.72733ZM7.22705 2.81824V1.81824H6.0037L6.24704 3.01714L7.22705 2.81824ZM2.77252 3.72733V4.72733H3.99594L3.75253 3.52837L2.77252 3.72733ZM2.7725 2.81824L3.75252 3.01715L3.99586 1.81824H2.7725V2.81824ZM6.27251 3.27273C6.27251 3.36122 6.26362 3.4466 6.24702 3.52837L8.20704 3.92628C8.25008 3.71428 8.27251 3.49561 8.27251 3.27273H6.27251ZM6.24704 3.01714C6.26363 3.0989 6.27251 3.18426 6.27251 3.27273H8.27251C8.27251 3.0499 8.25009 2.83128 8.20707 2.61933L6.24704 3.01714ZM7.22705 3.81824H19.5455V1.81824H7.22705V3.81824ZM19.5455 3.81824C19.2442 3.81824 19 3.57403 19 3.27278H21C21 2.46946 20.3488 1.81824 19.5455 1.81824V3.81824ZM19 3.27278C19 2.97154 19.2442 2.72733 19.5455 2.72733V4.72733C20.3488 4.72733 21 4.07611 21 3.27278H19ZM19.5455 2.72733H7.22703V4.72733H19.5455V2.72733ZM2.77252 2.72733H0.454546V4.72733H2.77252V2.72733ZM0.454546 2.72733C0.75579 2.72733 1 2.97153 1 3.27278H-1C-1 4.07611 -0.348775 4.72733 0.454546 4.72733V2.72733ZM1 3.27278C1 3.57403 0.755792 3.81824 0.454545 3.81824V1.81824C-0.348778 1.81824 -1 2.46946 -1 3.27278H1ZM0.454545 3.81824H2.7725V1.81824H0.454545V3.81824ZM3.72705 3.27273C3.72705 3.18426 3.73593 3.0989 3.75252 3.01715L1.79248 2.61933C1.74947 2.83128 1.72705 3.0499 1.72705 3.27273H3.72705ZM3.75253 3.52837C3.73593 3.4466 3.72705 3.36122 3.72705 3.27273H1.72705C1.72705 3.49561 1.74948 3.71428 1.79252 3.92628L3.75253 3.52837Z"
          fill="#120D2E"
          mask="url(#path-1-inside-1_1952_2686)"
        />
        <circle
          cx="4.99978"
          cy="3.27273"
          r="1.77273"
          stroke="#120D2E"
        />
        <path
          d="M16.7725 9.63637C16.7725 10.6154 15.9788 11.4091 14.9998 11.4091C14.0207 11.4091 13.2271 10.6154 13.2271 9.63637C13.2271 8.65732 14.0207 7.86365 14.9998 7.86365C15.9788 7.86365 16.7725 8.65732 16.7725 9.63637Z"
          stroke="#120D2E"
        />
        <path
          d="M11.3184 16C11.3184 16.9791 10.5247 17.7727 9.54568 17.7727C8.56663 17.7727 7.77295 16.9791 7.77295 16C7.77295 15.021 8.56663 14.2273 9.54568 14.2273C10.5247 14.2273 11.3184 15.021 11.3184 16Z"
          stroke="#120D2E"
        />
        <mask
          fill="white"
          id="path-6-inside-2_1952_2686"
        >
          <path
            clip-rule="evenodd"
            d="M17.2271 10.0909C17.2569 9.94401 17.2725 9.79202 17.2725 9.63637C17.2725 9.48068 17.2569 9.32865 17.227 9.18176H19.5455C19.7965 9.18176 20 9.38527 20 9.63631C20 9.88735 19.7965 10.0909 19.5455 10.0909H17.2271ZM12.7725 10.0909H0.454546C0.203508 10.0909 0 9.88735 0 9.63631C0 9.38527 0.203507 9.18176 0.454546 9.18176H12.7725C12.7427 9.32865 12.7271 9.48068 12.7271 9.63637C12.7271 9.79202 12.7427 9.94401 12.7725 10.0909Z"
            fill-rule="evenodd"
          />
        </mask>
        <path
          clip-rule="evenodd"
          d="M17.2271 10.0909C17.2569 9.94401 17.2725 9.79202 17.2725 9.63637C17.2725 9.48068 17.2569 9.32865 17.227 9.18176H19.5455C19.7965 9.18176 20 9.38527 20 9.63631C20 9.88735 19.7965 10.0909 19.5455 10.0909H17.2271ZM12.7725 10.0909H0.454546C0.203508 10.0909 0 9.88735 0 9.63631C0 9.38527 0.203507 9.18176 0.454546 9.18176H12.7725C12.7427 9.32865 12.7271 9.48068 12.7271 9.63637C12.7271 9.79202 12.7427 9.94401 12.7725 10.0909Z"
          fill="#120D2E"
          fill-rule="evenodd"
        />
        <path
          d="M17.2271 10.0909L16.247 9.89195L16.0037 11.0909H17.2271V10.0909ZM17.227 9.18176V8.18176H16.0036L16.247 9.38072L17.227 9.18176ZM12.7725 10.0909V11.0909H13.9958L13.7525 9.89195L12.7725 10.0909ZM12.7725 9.18176L13.7525 9.38072L13.9959 8.18176H12.7725V9.18176ZM16.2725 9.63637C16.2725 9.72484 16.2636 9.8102 16.247 9.89195L18.2071 10.2898C18.2501 10.0778 18.2725 9.8592 18.2725 9.63637H16.2725ZM16.247 9.38072C16.2636 9.4625 16.2725 9.54789 16.2725 9.63637H18.2725C18.2725 9.41348 18.2501 9.1948 18.207 8.9828L16.247 9.38072ZM17.227 10.1818H19.5455V8.18176H17.227V10.1818ZM19.5455 10.1818C19.2442 10.1818 19 9.93756 19 9.63631H21C21 8.83298 20.3488 8.18176 19.5455 8.18176V10.1818ZM19 9.63631C19 9.33506 19.2442 9.09085 19.5455 9.09085V11.0909C20.3488 11.0909 21 10.4396 21 9.63631H19ZM19.5455 9.09085H17.2271V11.0909H19.5455V9.09085ZM12.7725 9.09085H0.454546V11.0909H12.7725V9.09085ZM0.454546 9.09085C0.75579 9.09085 1 9.33506 1 9.63631H-1C-1 10.4396 -0.348775 11.0909 0.454546 11.0909V9.09085ZM1 9.63631C1 9.93755 0.755791 10.1818 0.454546 10.1818V8.18176C-0.348777 8.18176 -1 8.83298 -1 9.63631H1ZM0.454546 10.1818H12.7725V8.18176H0.454546V10.1818ZM13.7271 9.63637C13.7271 9.54788 13.7359 9.4625 13.7525 9.38072L11.7925 8.9828C11.7495 9.19481 11.7271 9.41349 11.7271 9.63637H13.7271ZM13.7525 9.89195C13.7359 9.8102 13.7271 9.72484 13.7271 9.63637H11.7271C11.7271 9.85919 11.7495 10.0778 11.7925 10.2898L13.7525 9.89195Z"
          fill="#120D2E"
          mask="url(#path-6-inside-2_1952_2686)"
        />
        <mask
          fill="white"
          id="path-8-inside-3_1952_2686"
        >
          <path
            clip-rule="evenodd"
            d="M11.773 16.4545C11.8028 16.3077 11.8184 16.1557 11.8184 16C11.8184 15.8443 11.8027 15.6923 11.7729 15.5454H19.5455C19.7965 15.5454 20 15.7489 20 16C20 16.251 19.7965 16.4545 19.5455 16.4545H11.773ZM7.3184 16.4545H0.454546C0.203508 16.4545 0 16.251 0 16C0 15.7489 0.203507 15.5454 0.454546 15.5454H7.31843C7.2886 15.6923 7.27295 15.8443 7.27295 16C7.27295 16.1557 7.28859 16.3077 7.3184 16.4545Z"
            fill-rule="evenodd"
          />
        </mask>
        <path
          clip-rule="evenodd"
          d="M11.773 16.4545C11.8028 16.3077 11.8184 16.1557 11.8184 16C11.8184 15.8443 11.8027 15.6923 11.7729 15.5454H19.5455C19.7965 15.5454 20 15.7489 20 16C20 16.251 19.7965 16.4545 19.5455 16.4545H11.773ZM7.3184 16.4545H0.454546C0.203508 16.4545 0 16.251 0 16C0 15.7489 0.203507 15.5454 0.454546 15.5454H7.31843C7.2886 15.6923 7.27295 15.8443 7.27295 16C7.27295 16.1557 7.28859 16.3077 7.3184 16.4545Z"
          fill="#120D2E"
          fill-rule="evenodd"
        />
        <path
          d="M11.773 16.4545L10.7929 16.2556L10.5496 17.4545H11.773V16.4545ZM11.7729 15.5454V14.5454H10.5495L10.7929 15.7444L11.7729 15.5454ZM7.3184 16.4545V17.4545H8.54174L8.29842 16.2556L7.3184 16.4545ZM7.31843 15.5454L8.29843 15.7444L8.54185 14.5454H7.31843V15.5454ZM10.8184 16C10.8184 16.0885 10.8095 16.1738 10.7929 16.2556L12.753 16.6534C12.796 16.4415 12.8184 16.2228 12.8184 16H10.8184ZM10.7929 15.7444C10.8095 15.8261 10.8184 15.9115 10.8184 16H12.8184C12.8184 15.7771 12.796 15.5585 12.7529 15.3464L10.7929 15.7444ZM11.7729 16.5454H19.5455V14.5454H11.7729V16.5454ZM19.5455 16.5454C19.2442 16.5454 19 16.3012 19 16H21C21 15.1966 20.3488 14.5454 19.5455 14.5454V16.5454ZM19 16C19 15.6987 19.2442 15.4545 19.5455 15.4545V17.4545C20.3488 17.4545 21 16.8033 21 16H19ZM19.5455 15.4545H11.773V17.4545H19.5455V15.4545ZM7.3184 15.4545H0.454546V17.4545H7.3184V15.4545ZM0.454546 15.4545C0.75579 15.4545 1 15.6987 1 16H-1C-1 16.8033 -0.348775 17.4545 0.454546 17.4545V15.4545ZM1 16C1 16.3012 0.755791 16.5454 0.454546 16.5454V14.5454C-0.348777 14.5454 -1 15.1966 -1 16H1ZM0.454546 16.5454H7.31843V14.5454H0.454546V16.5454ZM8.27295 16C8.27295 15.9115 8.28183 15.8261 8.29843 15.7444L6.33842 15.3464C6.29538 15.5585 6.27295 15.7771 6.27295 16H8.27295ZM8.29842 16.2556C8.28183 16.1738 8.27295 16.0885 8.27295 16H6.27295C6.27295 16.2228 6.29536 16.4415 6.33838 16.6534L8.29842 16.2556Z"
          fill="#120D2E"
          mask="url(#path-8-inside-3_1952_2686)"
        />
      </svg>
    </div>
    testLabel
  </button>
</div>
`;

exports[`IconButton IconButton has icon at the right 1`] = `
.emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row-reverse;
  -ms-flex-direction: row-reverse;
  flex-direction: row-reverse;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: space-around;
  -ms-flex-pack: space-around;
  -webkit-justify-content: space-around;
  justify-content: space-around;
  min-width: 7.5rem;
  height: 2rem;
  border-radius: 0.438rem;
  cursor: pointer;
  font-weight: 700;
  font-size: 0.813rem;
  background-color: #00DBC7;
  border: 0;
  color: #120D2E;
}

.emotion-0:hover {
  background-color: #00C6B3;
  border: 0;
  color: #120D2E;
}

.emotion-0:disabled {
  cursor: default;
  background-color: #EBEEF0;
  border: 0;
  color: #888596;
}

.emotion-1 {
  margin-top: 4px;
}

<div>
  <button
    class="emotion-0"
    disabled=""
  >
    <div
      class="emotion-1"
      data-testid="iconContainer"
    >
      <svg
        fill="none"
        height="20"
        viewBox="0 0 20 20"
        width="20"
        xmlns="http://www.w3.org/2000/svg"
      >
        <mask
          fill="white"
          id="path-1-inside-1_1952_2686"
        >
          <path
            clip-rule="evenodd"
            d="M7.22703 3.72733C7.25685 3.58044 7.27251 3.42841 7.27251 3.27273C7.27251 3.11708 7.25686 2.96509 7.22705 2.81824H19.5455C19.7965 2.81824 20 3.02174 20 3.27278C20 3.52382 19.7965 3.72733 19.5455 3.72733H7.22703ZM2.77252 3.72733H0.454546C0.203508 3.72733 0 3.52382 0 3.27278C0 3.02174 0.203507 2.81824 0.454545 2.81824H2.7725C2.7427 2.96509 2.72705 3.11708 2.72705 3.27273C2.72705 3.42841 2.7427 3.58044 2.77252 3.72733Z"
            fill-rule="evenodd"
          />
        </mask>
        <path
          clip-rule="evenodd"
          d="M7.22703 3.72733C7.25685 3.58044 7.27251 3.42841 7.27251 3.27273C7.27251 3.11708 7.25686 2.96509 7.22705 2.81824H19.5455C19.7965 2.81824 20 3.02174 20 3.27278C20 3.52382 19.7965 3.72733 19.5455 3.72733H7.22703ZM2.77252 3.72733H0.454546C0.203508 3.72733 0 3.52382 0 3.27278C0 3.02174 0.203507 2.81824 0.454545 2.81824H2.7725C2.7427 2.96509 2.72705 3.11708 2.72705 3.27273C2.72705 3.42841 2.7427 3.58044 2.77252 3.72733Z"
          fill="#120D2E"
          fill-rule="evenodd"
        />
        <path
          d="M7.22703 3.72733L6.24702 3.52837L6.00362 4.72733H7.22703V3.72733ZM7.22705 2.81824V1.81824H6.0037L6.24704 3.01714L7.22705 2.81824ZM2.77252 3.72733V4.72733H3.99594L3.75253 3.52837L2.77252 3.72733ZM2.7725 2.81824L3.75252 3.01715L3.99586 1.81824H2.7725V2.81824ZM6.27251 3.27273C6.27251 3.36122 6.26362 3.4466 6.24702 3.52837L8.20704 3.92628C8.25008 3.71428 8.27251 3.49561 8.27251 3.27273H6.27251ZM6.24704 3.01714C6.26363 3.0989 6.27251 3.18426 6.27251 3.27273H8.27251C8.27251 3.0499 8.25009 2.83128 8.20707 2.61933L6.24704 3.01714ZM7.22705 3.81824H19.5455V1.81824H7.22705V3.81824ZM19.5455 3.81824C19.2442 3.81824 19 3.57403 19 3.27278H21C21 2.46946 20.3488 1.81824 19.5455 1.81824V3.81824ZM19 3.27278C19 2.97154 19.2442 2.72733 19.5455 2.72733V4.72733C20.3488 4.72733 21 4.07611 21 3.27278H19ZM19.5455 2.72733H7.22703V4.72733H19.5455V2.72733ZM2.77252 2.72733H0.454546V4.72733H2.77252V2.72733ZM0.454546 2.72733C0.75579 2.72733 1 2.97153 1 3.27278H-1C-1 4.07611 -0.348775 4.72733 0.454546 4.72733V2.72733ZM1 3.27278C1 3.57403 0.755792 3.81824 0.454545 3.81824V1.81824C-0.348778 1.81824 -1 2.46946 -1 3.27278H1ZM0.454545 3.81824H2.7725V1.81824H0.454545V3.81824ZM3.72705 3.27273C3.72705 3.18426 3.73593 3.0989 3.75252 3.01715L1.79248 2.61933C1.74947 2.83128 1.72705 3.0499 1.72705 3.27273H3.72705ZM3.75253 3.52837C3.73593 3.4466 3.72705 3.36122 3.72705 3.27273H1.72705C1.72705 3.49561 1.74948 3.71428 1.79252 3.92628L3.75253 3.52837Z"
          fill="#120D2E"
          mask="url(#path-1-inside-1_1952_2686)"
        />
        <circle
          cx="4.99978"
          cy="3.27273"
          r="1.77273"
          stroke="#120D2E"
        />
        <path
          d="M16.7725 9.63637C16.7725 10.6154 15.9788 11.4091 14.9998 11.4091C14.0207 11.4091 13.2271 10.6154 13.2271 9.63637C13.2271 8.65732 14.0207 7.86365 14.9998 7.86365C15.9788 7.86365 16.7725 8.65732 16.7725 9.63637Z"
          stroke="#120D2E"
        />
        <path
          d="M11.3184 16C11.3184 16.9791 10.5247 17.7727 9.54568 17.7727C8.56663 17.7727 7.77295 16.9791 7.77295 16C7.77295 15.021 8.56663 14.2273 9.54568 14.2273C10.5247 14.2273 11.3184 15.021 11.3184 16Z"
          stroke="#120D2E"
        />
        <mask
          fill="white"
          id="path-6-inside-2_1952_2686"
        >
          <path
            clip-rule="evenodd"
            d="M17.2271 10.0909C17.2569 9.94401 17.2725 9.79202 17.2725 9.63637C17.2725 9.48068 17.2569 9.32865 17.227 9.18176H19.5455C19.7965 9.18176 20 9.38527 20 9.63631C20 9.88735 19.7965 10.0909 19.5455 10.0909H17.2271ZM12.7725 10.0909H0.454546C0.203508 10.0909 0 9.88735 0 9.63631C0 9.38527 0.203507 9.18176 0.454546 9.18176H12.7725C12.7427 9.32865 12.7271 9.48068 12.7271 9.63637C12.7271 9.79202 12.7427 9.94401 12.7725 10.0909Z"
            fill-rule="evenodd"
          />
        </mask>
        <path
          clip-rule="evenodd"
          d="M17.2271 10.0909C17.2569 9.94401 17.2725 9.79202 17.2725 9.63637C17.2725 9.48068 17.2569 9.32865 17.227 9.18176H19.5455C19.7965 9.18176 20 9.38527 20 9.63631C20 9.88735 19.7965 10.0909 19.5455 10.0909H17.2271ZM12.7725 10.0909H0.454546C0.203508 10.0909 0 9.88735 0 9.63631C0 9.38527 0.203507 9.18176 0.454546 9.18176H12.7725C12.7427 9.32865 12.7271 9.48068 12.7271 9.63637C12.7271 9.79202 12.7427 9.94401 12.7725 10.0909Z"
          fill="#120D2E"
          fill-rule="evenodd"
        />
        <path
          d="M17.2271 10.0909L16.247 9.89195L16.0037 11.0909H17.2271V10.0909ZM17.227 9.18176V8.18176H16.0036L16.247 9.38072L17.227 9.18176ZM12.7725 10.0909V11.0909H13.9958L13.7525 9.89195L12.7725 10.0909ZM12.7725 9.18176L13.7525 9.38072L13.9959 8.18176H12.7725V9.18176ZM16.2725 9.63637C16.2725 9.72484 16.2636 9.8102 16.247 9.89195L18.2071 10.2898C18.2501 10.0778 18.2725 9.8592 18.2725 9.63637H16.2725ZM16.247 9.38072C16.2636 9.4625 16.2725 9.54789 16.2725 9.63637H18.2725C18.2725 9.41348 18.2501 9.1948 18.207 8.9828L16.247 9.38072ZM17.227 10.1818H19.5455V8.18176H17.227V10.1818ZM19.5455 10.1818C19.2442 10.1818 19 9.93756 19 9.63631H21C21 8.83298 20.3488 8.18176 19.5455 8.18176V10.1818ZM19 9.63631C19 9.33506 19.2442 9.09085 19.5455 9.09085V11.0909C20.3488 11.0909 21 10.4396 21 9.63631H19ZM19.5455 9.09085H17.2271V11.0909H19.5455V9.09085ZM12.7725 9.09085H0.454546V11.0909H12.7725V9.09085ZM0.454546 9.09085C0.75579 9.09085 1 9.33506 1 9.63631H-1C-1 10.4396 -0.348775 11.0909 0.454546 11.0909V9.09085ZM1 9.63631C1 9.93755 0.755791 10.1818 0.454546 10.1818V8.18176C-0.348777 8.18176 -1 8.83298 -1 9.63631H1ZM0.454546 10.1818H12.7725V8.18176H0.454546V10.1818ZM13.7271 9.63637C13.7271 9.54788 13.7359 9.4625 13.7525 9.38072L11.7925 8.9828C11.7495 9.19481 11.7271 9.41349 11.7271 9.63637H13.7271ZM13.7525 9.89195C13.7359 9.8102 13.7271 9.72484 13.7271 9.63637H11.7271C11.7271 9.85919 11.7495 10.0778 11.7925 10.2898L13.7525 9.89195Z"
          fill="#120D2E"
          mask="url(#path-6-inside-2_1952_2686)"
        />
        <mask
          fill="white"
          id="path-8-inside-3_1952_2686"
        >
          <path
            clip-rule="evenodd"
            d="M11.773 16.4545C11.8028 16.3077 11.8184 16.1557 11.8184 16C11.8184 15.8443 11.8027 15.6923 11.7729 15.5454H19.5455C19.7965 15.5454 20 15.7489 20 16C20 16.251 19.7965 16.4545 19.5455 16.4545H11.773ZM7.3184 16.4545H0.454546C0.203508 16.4545 0 16.251 0 16C0 15.7489 0.203507 15.5454 0.454546 15.5454H7.31843C7.2886 15.6923 7.27295 15.8443 7.27295 16C7.27295 16.1557 7.28859 16.3077 7.3184 16.4545Z"
            fill-rule="evenodd"
          />
        </mask>
        <path
          clip-rule="evenodd"
          d="M11.773 16.4545C11.8028 16.3077 11.8184 16.1557 11.8184 16C11.8184 15.8443 11.8027 15.6923 11.7729 15.5454H19.5455C19.7965 15.5454 20 15.7489 20 16C20 16.251 19.7965 16.4545 19.5455 16.4545H11.773ZM7.3184 16.4545H0.454546C0.203508 16.4545 0 16.251 0 16C0 15.7489 0.203507 15.5454 0.454546 15.5454H7.31843C7.2886 15.6923 7.27295 15.8443 7.27295 16C7.27295 16.1557 7.28859 16.3077 7.3184 16.4545Z"
          fill="#120D2E"
          fill-rule="evenodd"
        />
        <path
          d="M11.773 16.4545L10.7929 16.2556L10.5496 17.4545H11.773V16.4545ZM11.7729 15.5454V14.5454H10.5495L10.7929 15.7444L11.7729 15.5454ZM7.3184 16.4545V17.4545H8.54174L8.29842 16.2556L7.3184 16.4545ZM7.31843 15.5454L8.29843 15.7444L8.54185 14.5454H7.31843V15.5454ZM10.8184 16C10.8184 16.0885 10.8095 16.1738 10.7929 16.2556L12.753 16.6534C12.796 16.4415 12.8184 16.2228 12.8184 16H10.8184ZM10.7929 15.7444C10.8095 15.8261 10.8184 15.9115 10.8184 16H12.8184C12.8184 15.7771 12.796 15.5585 12.7529 15.3464L10.7929 15.7444ZM11.7729 16.5454H19.5455V14.5454H11.7729V16.5454ZM19.5455 16.5454C19.2442 16.5454 19 16.3012 19 16H21C21 15.1966 20.3488 14.5454 19.5455 14.5454V16.5454ZM19 16C19 15.6987 19.2442 15.4545 19.5455 15.4545V17.4545C20.3488 17.4545 21 16.8033 21 16H19ZM19.5455 15.4545H11.773V17.4545H19.5455V15.4545ZM7.3184 15.4545H0.454546V17.4545H7.3184V15.4545ZM0.454546 15.4545C0.75579 15.4545 1 15.6987 1 16H-1C-1 16.8033 -0.348775 17.4545 0.454546 17.4545V15.4545ZM1 16C1 16.3012 0.755791 16.5454 0.454546 16.5454V14.5454C-0.348777 14.5454 -1 15.1966 -1 16H1ZM0.454546 16.5454H7.31843V14.5454H0.454546V16.5454ZM8.27295 16C8.27295 15.9115 8.28183 15.8261 8.29843 15.7444L6.33842 15.3464C6.29538 15.5585 6.27295 15.7771 6.27295 16H8.27295ZM8.29842 16.2556C8.28183 16.1738 8.27295 16.0885 8.27295 16H6.27295C6.27295 16.2228 6.29536 16.4415 6.33838 16.6534L8.29842 16.2556Z"
          fill="#120D2E"
          mask="url(#path-8-inside-3_1952_2686)"
        />
      </svg>
    </div>
    testLabel
  </button>
</div>
`;
